src/status_im2/contexts/quo_preview/share/share_qr_code.cljs

```diff
@@ -1,34 +1,141 @@
 (ns status-im2.contexts.quo-preview.share.share-qr-code
-  (:require
-    [quo.core :as quo]
-    [reagent.core :as reagent]
-    [status-im2.contexts.quo-preview.preview :as preview]
-    [utils.image-server :as image-server]
-    [utils.re-frame :as rf]))
+  (:require [quo.core :as quo]
+            [react-native.core :as rn]
+            [reagent.core :as reagent]
+            [status-im2.common.resources :as resources]
+            [status-im2.contexts.quo-preview.preview :as preview]
+            [utils.image-server :as image-server]
+            [utils.re-frame :as rf]))
 
 (def descriptor
-  [{:key :url :type :text}
-   {:key :link-title :type :text}])
+  [{:key   :qr-data
+    :label "QR data:"
+    :type  :text}
+   {:key     :type
+    :type    :select
+    :options [{:key :profile}
+              {:key :wallet-legacy}
+              {:key :wallet-multichain}]}])
+
+(def profile-descriptor
+  [{:key     :profile-picture
+    :type    :select
+    :options [{:key   (resources/get-mock-image :user-picture-female2)
+               :value "User 1"}
+              {:key   (resources/get-mock-image :user-picture-male4)
+               :value "User 2"}
+              {:key   nil
+               :value "No picture"}]}
+   {:key  :full-name
+    :type :text}
+   (preview/customization-color-option)])
+
+(def wallet-legacy-descriptor
+  [{:key     :emoji
+    :type    :select
+    :options [{:key "🐈"}
+              {:key "👻"}
+              {:key "🐧"}]}
+   (preview/customization-color-option)])
+
+(def possible-networks [:ethereum :optimism :arbitrum :myNet])
+
+(def wallet-multichain-descriptor
+  [{:key     :emoji
+    :type    :select
+    :options [{:key "🐈"}
+              {:key "👻"}
+              {:key "🐧"}]}
+   (preview/customization-color-option)
+   {:key     :networks
+    :type    :select
+    :options [{:key   (take 1 possible-networks)
+               :value "Ethereum"}
+              {:key   (take 2 possible-networks)
+               :value "Ethereum and Optimism"}
+              {:key   (take 3 possible-networks)
+               :value "Ethereum, Optimism and Arbitrum"}
+              {:key   (take 4 possible-networks)
+               :value "Ethereum, Optimism, Arbitrum and unknown"}]}])
+
+(defn- get-network-short-name-url
+  [network-kw]
```

Comment:
Just a bit unconventional to add types to names in Clojure.

Reply:
undone!